src/bitcoin-cli.cpp

```diff
@@ -291,6 +293,184 @@ class GetinfoRequestHandler: public BaseRequestHandler
     }
 };
 
+/** Process netinfo requests */
+class NetinfoRequestHandler : public BaseRequestHandler
+{
+private:
+    bool IsAddrIPv6(const std::string& addr) const
+    {
+        return !addr.empty() && addr.front() == '[';
+    }
+    bool IsInboundOnion(int mapped_as, const std::string& addr, const std::string& addr_local) const
+    {
+        return mapped_as == 0 && addr.find("127.0.0.1") == 0 && addr_local.find(".onion") != std::string::npos;
+    }
+    bool IsOutboundOnion(const std::string& addr) const
+    {
+        return addr.find(".onion") != std::string::npos;
+    }
+    bool m_verbose{false}; //!< Whether user requested verbose -netinfo report
+
+    enum struct m_conn_type {
+        ipv4,
+        ipv6,
+        onion,
+    };
+
+    struct m_peer {
+        int id;
+        int mapped_as;
+        int version;
+        int64_t conn_time;
+        int64_t last_recv;
+        int64_t last_send;
+        double min_ping;
+        double ping;
+        std::string addr;
+        std::string sub_version;
+        m_conn_type conn_type;
+        bool is_block_relay;
+        bool is_outbound;
+        bool operator<(const m_peer& rhs) const { return std::tie(is_outbound, min_ping) < std::tie(rhs.is_outbound, rhs.min_ping); }
+    };
+
+    std::string ConnTypeEnumToString(m_conn_type t)
+    {
+        switch (t) {
+        case m_conn_type::ipv4: return "ipv4";
+        case m_conn_type::ipv6: return "ipv6";
+        case m_conn_type::onion: return "onion";
+        } // no default case, so the compiler can warn about missing cases
+        assert(false);
+    }
+
+public:
+    const int ID_PEERINFO = 0;
+    const int ID_NETWORKINFO = 1;
+
+    UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) override
+    {
+        if (!args.empty()) {
+            const std::string arg{ToLower(args.at(0))};
+            m_verbose = (arg == "true" || arg == "t");
```

Comment:
This uses a different boolean argument parsing than the rest of the software, which doesn't interpret "1" as true, for example. And it doesn't raise a parse error for invalid values, but simply ignores them (and assumes them to be false). 
I understand that it's not possible to use `GetBoolArg` here because the arguments are passed in, but it'd be nice to be consistent.

Alternatively we could make this optional argument "verbose" instead of "true", to not confuse it with boolean parsing at all. I don't know.

Reply:
Yes, this is a new context. As this is very simple, doesn't wrap any RPC commands and will only be used by humans, I began with the simplest and probably least annoying option to users of not raising parsing errors.

Reply:
I originally counted "1" as truthy but removed that after https://github.com/bitcoin/bitcoin/pull/19643#discussion_r468068980.

Reply:
I'm considering proposing replacing the boolean arg with an integer one in the follow-up, to allow choosing from a number of degrees of detail rather than just two.